contracts/Registrar.sol

```diff
@@ -6,86 +6,157 @@ import "@ensdomains/ens/contracts/ENS.sol";
+    }
+
+    function commit(bytes32 commitment) external auth {
+        commited[commitment] = block.number;
+    }
+
+    function mkCommitment(
+        string calldata name, address owner, uint salt
+    ) external pure returns (bytes32) {
+        return keccak256(abi.encodePacked(name, owner, salt));
+    }
+}
+
 contract Registrar {
-    /// The ENS registry.
-    ENS public immutable ens;
 
-    /// The price oracle.
-    address public oracleAddress;
+    // --- DATA ---
 
-    /// The Rad/Eth exchange.
-    address public exchangeAddress;
+    /// The ENS registry.
+    ENS public ens;
 
     /// The Radicle ERC20 token.
     ERC20Burnable public immutable rad;
 
+    /// The commitment storage contract
+    Commitments public immutable commitments = new Commitments();
+
+    /// The namehash of the `eth` TLD in the ENS registry, eg. namehash("eth").
+    bytes32 public constant ethNode = keccak256(abi.encodePacked(bytes32(0), keccak256("eth")));
+
     /// The namehash of the node in the `eth` TLD, eg. namehash("radicle.eth").
-    bytes32 public immutable domain;
+    bytes32 public constant radNode = keccak256(abi.encodePacked(ethNode, keccak256("radicle")));
 
     /// The token ID for the node in the `eth` TLD, eg. sha256("radicle").
-    uint256 public immutable tokenId;
+    uint256 public constant tokenId = uint(keccak256(abi.encodePacked("radicle")));
 
-    /// Registration fee in *USD*.
-    uint256 public registrationFeeUsd = 10;
+    /// The minimum number of blocks that must have passed between a commitment and name registration
+    // TODO: justify this as a default value...
+    uint256 public minCommitmentAge = 50;
 
     /// Registration fee in *Radicle* (uRads).
-    uint256 public registrationFeeRad = 1e18;
+    uint256 public fee = 1e18;
 
-    /// The contract admin who can set fees.
-    address public admin;
+    // --- LOGS ---
 
     /// @notice A name was registered.
-    event NameRegistered(bytes32 indexed label, address indexed owner);
+    event NameRegistered(string indexed name, bytes32 indexed label, address indexed owner);
+
+    /// @notice A commitment was made
+    event CommitmentMade(bytes32 commitment, uint blockNumber);
+
+    /// @notice The contract admin was changed
+    event AdminChanged(address newAdmin);
+
+    /// @notice The registration fee was changed
+    event RegistrationFeeChanged(uint amt);
+
+    /// @notice The ownership of the domain was changed
+    event DomainOwnershipChanged(address newOwner);
+
+    /// @notice The registration fee was changed
+    event ResolverChanged(address resolver);
+
+    /// @notice The registration fee was changed
+    event TTLChanged(uint64 amt);
+
+    /// @notice The ens registry was updated
+    event EnsChanged(address ens);
+
+    /// @notice The minimum age for a commitment was changed
+    event MinCommitmentAgeChanged(uint256 amt);
+
+    // -- MATH --
+
+    function add(uint x, uint y) internal pure returns (uint z) {
```

Comment:
unused, and seems fine to use unsafe math in the blocknumber calculation later

Reply:
nice catch, thx :)